### PR TITLE
feat(onyx-356): single activity panel notification screen

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-356]

### Description

- [Figma prototype](https://www.figma.com/proto/VjFsvmJ7e5geXuRGDeKRCg/Onyx---Design-Strategy-WIP?page-id=841%3A3070&type=design&node-id=1187-4428&viewport=99%2C-1064%2C0.31&t=2zHDojRSyxBIT0wv-1&scaling=scale-down&starting-point-node-id=851%3A859)

I've added a new screen for individual activity panel notifications. You can navigate to it by clicking on an activity panel item from activity panel or latest activities rail.

### Demo

https://github.com/artsy/eigen/assets/3934579/cee75832-aa92-4045-8a67-71e2ade1b4da

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
  - [ ] I will appreciate if someone tests my changes on Android simulator
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
  - [x] This is not a final version - thus no tests so far 
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- behind feature flag: new screen for individual activity panel notifications - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-356]: https://artsyproduct.atlassian.net/browse/ONYX-356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (behind feature flag: new screen for individual activity panel notifications - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7068fc8f8e3a231b543486c86cb3a4f216282fee
</p>

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      <Box>
+        <Flex flexDirection="row">
+          <ArtworkImage
+            imageURL={imageURLs[0]}
+            size={rowImageWidth}
+            artworkHref={artworkNodes[0].href!}
+          />
+          <Spacer x={`${IMAGE_OFFSET}px`} />
+
+          {artworkNodes.length > 1 && (
```
Comment (dblandin) [inline src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:44]: **suggestion / non-blocking**: I'm not super familiar implementing this kind of grid in Eigen but I wonder if there's an opportunity to use some of React Native's support for [flex item wrapping](https://reactnative.dev/docs/flexbox#flex-wrap) and perhaps also setting a [gap gutter](https://reactnative.dev/docs/flexbox#row-gap-column-gap-and-gap) to achieve the same layout.

Happy to merge as is and continue iterating, as this is still under a feature flag, but maybe something to look into.

**Review by dblandin — APPROVED** [review]
LGTM overall!

File: src/app/Scenes/Activity/ActivityItem.tsx [hunk 2]
```diff
@@ -21,6 +23,7 @@ const UNREAD_INDICATOR_SIZE = 8
 const ARTWORK_IMAGE_SIZE = 55
 
 export const ActivityItem: React.FC<ActivityItemProps> = (props) => {
+  const navigateToASingleNotification = useFeatureFlag("AREnableSingleActivityPanelScreen")
```
Comment (MounirDhahri) [inline src/app/Scenes/Activity/ActivityItem.tsx:26]:
```suggestion
  const enableNavigateToASingleNotification = useFeatureFlag("AREnableSingleActivityPanelScreen")
```

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  return (
+    <>
+      <Box>
+        <Flex flexDirection="row">
+          <ArtworkImage
+            imageURL={imageURLs[0]}
+            size={rowImageWidth}
+            artworkHref={artworkNodes[0].href!}
+          />
+          <Spacer x={`${IMAGE_OFFSET}px`} />
```
Comment (MounirDhahri) [inline src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:42]: **Q:** any reason for not using `x={1}` here?

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:44]: i would suggest looking into this as well

File: src/app/Scenes/Activity/ActivityItemScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  query ActivityItemScreenQuery($internalID: String!) {
+    me {
+      notification(id: $internalID) {
+        title
+        message
+        notificationType
+        publishedAt(format: "RELATIVE")
+        targetHref
+
+        ...ActivityItemArtworksGrid_notification
```
Comment (MounirDhahri) [inline src/app/Scenes/Activity/ActivityItemScreen.tsx:133]:
```suggestion
        targetHref
        ...ActivityItemArtworksGrid_notification
```

**Review by MounirDhahri — APPROVED** [review]
LGTM - nice PR!

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx (hunk 3 above)
Comment (nickskalkin) [reply src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:42]: My thinking was that I have a constant IMAGE_OFFSET (in pixels), and I do all math in this component based on this constant, for instance `rowImageWidth` is calculated like this: `(width - (TOTAL_HORIZONTAL_PADDING + IMAGE_OFFSET)) / 2`.

If I were to change IMAGE_OFFSET to 15 pixels, for instance,  nothing would be broken

Comment (nickskalkin) [reply src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:42]: but I'll try to rewrite with Flex gap instead, so never mind 😄

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx (hunk 1 above)
Comment (nickskalkin) [reply src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:44]: Yeah, this whole component looks weird. I took [this one](https://github.com/artsy/eigen/blob/main/src/app/Scenes/ArtworkLists/FourUpImageLayout.tsx) as for basis, but in current case we don't need such grid. I'll try to rewrite with Flex gap gutter

File: src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/app/Scenes/Activity/ActivityItemArtworksGrid.tsx:42]: thanks for explaining 🙏

**Review by MounirDhahri — APPROVED** [review]
Thanks Nikita 🙏

**Pull request merged**